test/test_eager_transforms.py

```diff
@@ -1255,18 +1255,10 @@ def f(x):
 
         x = torch.randn(3, device=device)
 
-        # TODO: Remove when https://github.com/pytorch/functorch/issues/392
-        # is fixed
-        if jacapi == jacrev:
-            result, aux = jacapi(f, has_aux=True)(x)
-            self.assertEqual(result, torch.eye(3, 3, device=device))
-            expected_aux = {'a': x.cos(), 'b': [x.tan()]}
-            self.assertEqual(aux, expected_aux)
-        else:
-            result, aux = jacapi(f)(x)
-            self.assertEqual(result, torch.eye(3, 3, device=device))
-            expected_aux = {'a': -torch.diag(x.sin()), 'b': [torch.diag(1.0 + x.tan() ** 2)]}
-            self.assertEqual(aux, expected_aux)
+        result, aux = jacapi(f, has_aux=True)(x)
+        self.assertEqual(result, torch.eye(3, 3, device=device))
+        expected_aux = {'a': x.cos(), 'b': [x.tan()]}
+        self.assertEqual(aux, expected_aux)
```

Comment:
Could we also add a test for jacapi that checks the error cases? E.g. if aux is a number, then the current state is that we error out. It'll be good to also check the error message for that is nice.